packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -0,0 +1,368 @@
+import cdk = require('@aws-cdk/cdk');
+
+import { CfnMesh } from './appmesh.generated';
+import { ListenerProps, NAME_TAG } from './shared-interfaces';
+import { VirtualNode, VirtualNodeBackendProps } from './virtual-node';
+import { VirtualRouter, VirtualRouterBaseProps } from './virtual-router';
+import { VirtualService, VirtualServiceBaseProps } from './virtual-service';
+
+/**
+ * These properties are used when adding a VirtualNode through the Mesh type
+ */
+export interface AddVirtualNodeProps {
+  /**
+   * The name of the VirtualNode, only used as identifier
+   *
+   * @default optional if not provided, the id property will be used
+   */
+  readonly nodeName?: string;
+  /**
+   * The hostname for which to identify the VirtualNode, NOT the full FQDN
+   *
+   * @example serviceb NOT serviceb.domain.local
+   */
+  readonly hostname: string;
+  /**
+   * The service discovery namespace name
+   * @example domain.local
+   */
+  readonly namespaceName: string;
+  /**
+   * The backend services this node expects to send traffic to
+   *
+   * @default none
+   */
+  readonly backends?: VirtualNodeBackendProps[];
+  /**
+   * @default none
+   * if not specified must call addListeners(), addPortMappings() or addPortAndHealthCheckMappings()
+   */
+  readonly listeners?: ListenerProps;
+}
+
+/**
+ * A utility enum created for the egressFilter type property, the default of DROP_ALL,
+ * allows traffic only to other resources inside the mesh, or API calls to amazon resources.
+ *
+ * @default Drop_All
+ */
+export enum MeshFilterType {
+  /**
+   * Allows all outbound traffic
+   */
+  Allow_All = 'ALLOW_ALL',
+  /**
+   * Allows traffic only to other resources inside the mesh, or API calls to amazon resources
+   */
+  Drop_All = 'DROP_ALL',
+}
+
+/**
+ * Used to populate the spec property for the mesh, only available option today is egressFilter
+ *
+ */
+export interface MeshSpec {
+  /**
+   * Egress filter to be applied to the Mesh
+   *
+   * @default DROP_ALL
+   */
+  readonly egressFilter?: MeshFilterType;
+}
+
+/**
+ * Used for importing and exporting Mesh(es)
+ */
+export interface MeshImportProps {
+  /**
+   * The AppMesh name to import
+   */
+  readonly meshName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  readonly meshArn?: string;
+  /**
+   * The unique identifier for the mesh
+   */
+  readonly meshUid: string;
+}
+
+/**
+ * Interface wich all Mesh based classes MUST implement
+ */
+export interface IMesh extends cdk.IConstruct {
+  /**
+   * The name of the AppMesh mesh
+   */
+  readonly meshName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  readonly meshArn?: string;
```

Comment:
How can an `AppMesh` *not* have an `ARN`?

Reply:
This was already fixed, leftover from old testing just hadn't committed to PR yet.. will show fixed on next one.